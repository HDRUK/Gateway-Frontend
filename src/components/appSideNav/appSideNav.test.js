import React from "react";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import AppSideNav from "./appSideNav.js";

describe("<AppSideNav> ", () => {
    let renderedComponent;

    beforeEach(() => {
        renderedComponent = create(
            <BrowserRouter initialEntries={["/search"]}>
                <AppSideNav />
            </BrowserRouter>
        );
    });

    it("should test the AppSideNav Snapshot", () => {
        expect(renderedComponent).toMatchSnapshot();
    });

    // Is this kind of test required / how do we implement this?
    // it("should switch content to about when clicking 'about' link", () => {
    //     console.log(history, location);
    //     const aboutLink = renderedComponent.root.findByProps({ href: "/about" });
    //     act(() => aboutLink.props.navigate());
    //     console.log(history, location);

    //     // expect(history.location.pathname).toBe("/about");
    // });
});
