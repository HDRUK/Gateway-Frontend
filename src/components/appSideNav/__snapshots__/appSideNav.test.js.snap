// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppSideNav>  should test the AppSideNav Snapshot 1`] = `
Array [
  <div
    className="bx--side-nav__overlay bx--side-nav__overlay-active"
  />,
  <nav
    aria-label="Side navigation"
    className="bx--side-nav__navigation bx--side-nav bx--side-nav--expanded sc-kEYyzF exvBSO"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <p
      className="sc-gqjmRU bvaZHa"
    >
      Nicola Blackwood
    </p>
    <p
      className="sc-VigVT edmemT"
    >
      UK Government
    </p>
    <div
      className="sc-fjdhpX QuAmw"
    />
    <ul
      className="bx--side-nav__items"
    >
      <a
        className="sc-kgoBCf eeeUOF"
        href="/search"
        onClick={[Function]}
      >
        <li
          className="bx--side-nav__item"
        >
          <a
            className="bx--side-nav__link"
          >
            <span
              className="bx--side-nav__link-text"
            >
              Search
            </span>
          </a>
        </li>
      </a>
      <a
        className="sc-kgoBCf eeeUOF"
        href="/my-searches"
        onClick={[Function]}
      >
        <li
          className="bx--side-nav__item"
        >
          <a
            className="bx--side-nav__link"
          >
            <span
              className="bx--side-nav__link-text"
            >
              My searches
            </span>
          </a>
        </li>
      </a>
      <a
        className="sc-kgoBCf eeeUOF"
        href="/browse"
        onClick={[Function]}
      >
        <li
          className="bx--side-nav__item"
        >
          <a
            className="bx--side-nav__link"
          >
            <span
              className="bx--side-nav__link-text"
            >
              Browse
            </span>
          </a>
        </li>
      </a>
      <a
        className="sc-kgoBCf eeeUOF"
        href="/about"
        onClick={[Function]}
      >
        <li
          className="bx--side-nav__item"
        >
          <a
            className="bx--side-nav__link"
          >
            <span
              className="bx--side-nav__link-text"
            >
              About
            </span>
          </a>
        </li>
      </a>
      <a
        className="sc-kgoBCf eeeUOF"
        href="/help"
        onClick={[Function]}
      >
        <li
          className="bx--side-nav__item"
        >
          <a
            className="bx--side-nav__link"
          >
            <span
              className="bx--side-nav__link-text"
            >
              Help
            </span>
          </a>
        </li>
      </a>
    </ul>
  </nav>,
]
`;
